src/routes/messages/messages.controller.spec.ts

```diff
@@ -603,4 +604,80 @@ describe('Messages controller', () => {
         });
     });
   });
+
+  describe('Create messages', () => {
+    it('Success', async () => {
+      const chain = chainBuilder().build();
+      const safe = safeBuilder().build();
+      const message = messageBuilder()
+        .with('safeAppId', null)
+        .with('created', faker.date.recent())
+        .build();
+      mockNetworkService.get.mockImplementation((url) =>
+        url === `${safeConfigUrl}/api/v1/chains/${chain.chainId}`
+          ? Promise.resolve({ data: chain })
+          : Promise.reject(`No matching rule for url: ${url}`),
+      );
+      mockNetworkService.post.mockImplementation((url) =>
+        url ===
+        `${chain.transactionService}/api/v1/safes/${safe.address}/messages/`
+          ? Promise.resolve({ data: messageToJson(message) })
+          : Promise.reject(`No matching rule for url: ${url}`),
+      );
+
+      await request(app.getHttpServer())
+        .post(`/v1/chains/${chain.chainId}/safes/${safe.address}/messages`)
+        .send(createMessageDtoBuilder().build())
+        .expect(200)
+        .expect(JSON.stringify(messageToJson(message)));
+    });
+
+    it('should return an error from the provider', async () => {
```

Comment:
What does "provider" represent? My suggestion is to use more specific terms. Many entities can be seen as "providers" and for the reader it can get confusing.

Reply:
I was trying to be agnostic as today some entity could be provided by the Transaction Service, Config Service, etc. but in the future it could change. That's why I was using the term "provider". But yes, maybe we can just use the concrete references, I agree 🙂

Reply:
I was trying to be agnostic as today some entities could be provided by the Transaction Service, Config Service, etc. but in the future, it could change. That's why I was using the term "provider". But yes, maybe we can just use the concrete references, I agree 🙂

Changed in [8fab600](https://github.com/5afe/safe-client-gateway-nest/pull/281/commits/8fab60003b58dbff0be57a86d59f439e5af48451)

Reply:
> I was trying to be agnostic as today some entities could be provided by the Transaction Service, Config Service, etc. but in the future, it could change. That's why I was using the term "provider". But yes, maybe we can just use the concrete references, I agree 🙂
> 
> Changed in [8fab600](https://github.com/5afe/safe-client-gateway-nest/pull/281/commits/8fab60003b58dbff0be57a86d59f439e5af48451)

I think in this case we can be more specific because in my opinion it makes the test easier to read. To answer questions like: "which service returns the error in this scenario?"